allocator/allocator.go

```diff
@@ -225,19 +206,44 @@ func removeDeadAssignments(txn checkpointTxn, ks *keyspace.KeySpace, asn keyspac
 	return nil
 }
 
+func solveDesiredAssignments(s *State, desired []Assignment) []Assignment {
+	// Number of items to lump into each invocation of push/relabel.
+	// This is an arbitrary number which is empirically fast to solve,
+	// but is large enough that we're unlikely to see further improvements
+	// from finding a global maximum flow.
+	// Splitting the problem in this way makes it big-O linear on items,
+	// instead of combinatorial over members and items.
+	const itemsPerNetwork = 10000
+
+	// NOTE(johnny): this could trivially be parallelized if needed.
```

Comment:
Answered below.

Then re: parallelization, `State` is unchanged when building / evaluating the max flow network, so multiple goroutines can work in parallel. I didn't do it yet because it's extra SLOC we don't appear to need, and I didn't know for-sure that the memory impact would be negligible.